src/core/Oscillator.cpp

```diff
@@ -88,6 +101,211 @@ void Oscillator::update( sampleFrame * _ab, const fpp_t _frames,
+	}
+}
+
+void Oscillator::generateSawWaveTable(int bands, sample_t * table)
+{
+	// sawtooth wave contain both even and odd harmonics
+	// hence sinewaves are added for all bands
+	// https://en.wikipedia.org/wiki/Sawtooth_wave
+	float max = 0;
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; i++)
+	{
+		table[i] = 0.0;
+		for (float n = 1; n <= bands; n++)
+		{
+			table[i] += ((int)n%2 ? 1.0f : -1.0f) / n * sinf(F_2PI * i * n / (float)OscillatorConstants::WAVETABLE_LENGTH);
+		}
+		max = fmax(max, table[i]);
+	}
+
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; i++)
+	{
+		table[i] /= max;
+	}
+}
+
+
+void Oscillator::generateTriangleWaveTable(int bands, sample_t * table)
+{
+	// triangle waves contain only odd harmonics
+	// hence sinewaves are added for alternate bands
+	// https://en.wikipedia.org/wiki/Triangle_wave
+	float max = 0;
+	for (int i = 0 ; i < OscillatorConstants::WAVETABLE_LENGTH; i++)
+	{
+		table[i] = 0.0;
+		for (float n = 0; n <= bands * 0.5; n++)
+		{
+			table[i] += ((int)n%2 ? -1.0f : +1.0f) / powf((float)(2 * n + 1),
+							   2.0f) *
+					sinf(F_2PI * (2.0f * n + 1) * i / (float)OscillatorConstants::WAVETABLE_LENGTH);
+		}
+		max = fmax(max, table[i]);
+	}
+
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; i++)
+	{
+		table[i] /= max;
+	}
+}
+
+void Oscillator::generateSquareWaveTable(int bands, sample_t * table)
+{
+	// square waves only contain odd harmonics,
+	// at diffrent levels when compared to triangle waves
+	// https://en.wikipedia.org/wiki/Square_wave
+	float max = 0;
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; i++)
+	{
+		table[i] = 0.0f;
+		for (float n = 1; n <= bands; n += 2)
+		{
+			table[i] += (1.0f / n) * sinf(F_2PI * i * n / OscillatorConstants::WAVETABLE_LENGTH);
+		}
+		max = fmax(max, table[i]);
+	}
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; i++)
+	{
+		table[i] /= max;
+	}
+}
+
+
+
+//expects sample in sample buffer
+void Oscillator::generateFromFFT(int bands, float threshold, sample_t * table)
+{
+	float max = 0;
+	
+	//set unrequired bands to zero 
+	for (int i = bands; i < OscillatorConstants::WAVETABLE_LENGTH * 2 + 1 - bands; i++)
+	{
+		s_specBuf[i][0] = 0.0f;
+		s_specBuf[i][1] = 0.0f;
+	}
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH * 2 + 1; ++i)
+	{
+		if (s_specBuf[i][0] * s_specBuf[i][0] + s_specBuf[i][1] * s_specBuf[i][1] < threshold)
+		{
+			s_specBuf[i][0] = 0.0f;
+			s_specBuf[i][1] = 0.0f;
+		}
+	}
+	//ifft
+	fftwf_execute(s_ifftPlan);
+	//normalise
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; ++i)
+	{
+		max = fmax(max, s_sampleBuffer[i]);
+	}
+	//copy to generateWaveTable
+	for (int i = 0; i < OscillatorConstants::WAVETABLE_LENGTH; ++i)
+	{
+		table[i] = s_sampleBuffer[i] / max;
+	}
+}
+
+void Oscillator::generateAntiAliasUserWaveTable(SampleBuffer *sampleBuffer)
+{
+	delete sampleBuffer->m_userAntiAliasWaveTable;
```

Comment:
Possible memory leaks: `sampleBuffer->m_userAntiAliasWaveTable[i]`(`0 <= i < WAVE_TABLES_PER_WAVEFORM_COUNT`) are not deleted.

I also think you can move allocation/deallocation code to `SampleBuffer` instead. If so, you can share the deletion code with `SampleBuffer` destructor.
Alternatively, you may use `std::unique_ptr` as suggested by @Wallacoloo. I can provide an example code if you want.